configuration/src/main/java/pl/edu/uj/kimage/configuration/ConfigurationReader.java

```diff
@@ -0,0 +1,84 @@
+package pl.edu.uj.kimage.configuration;
+
+import com.google.common.collect.ImmutableMap;
+
+import javax.annotation.concurrent.ThreadSafe;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.*;
+import java.util.List;
+
+@ThreadSafe
+public class ConfigurationReader {
+    private static final int KEY_INDEX = 0;
+    private static final int VALUE_INDEX = 1;
+    private static final String SEPARATOR = "=";
+    private static ConfigurationReader INSTANCE;
+    private static ImmutableMap<String, String> SETTINGS;
+
+    private ConfigurationReader(String path) {
+        SETTINGS = loadConfig(path);
+        registerWatchService(path);
+    }
+
+    /**
+     * @param path path to properties file in resource folder
+     */
+    public static ConfigurationReader getInstance(String path) {
+        if (INSTANCE == null) {
+            synchronized (ConfigurationReader.class) {
+                if (INSTANCE == null) {
+                    INSTANCE = new ConfigurationReader(path);
+                }
+            }
+        }
+        return INSTANCE;
+    }
+
+    ImmutableMap<String, String> loadConfig(String fileName) {
+        ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+        try {
+            Path path = Paths.get(ClassLoader.getSystemResource(fileName).toURI());
```

Comment:
Do we really want to load it from classpath? We were discussing loading file from working directory and if it is not present then we are looking for it in classpath. If loaded from classpath I would say that watch service is not necessary.